utils/xml_memory_utils.py

```
@@ -0,0 +1,257 @@
+    root.set('message_count', str(len(messages)))
+    
+    for msg in messages:
+        msg_elem = message_to_xml_element(msg)
+        root.append(msg_elem)
+    
+    # 转换为字符串
+    xml_str = ET.tostring(root, encoding='unicode')
+    
+    if pretty_print:
+        # 美化输出
+        dom = minidom.parseString(xml_str)
```

Ruff (0.14.8)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S318)